metric_tracking/__init__.py

```diff
@@ -0,0 +1,212 @@
+import os
+import sys 
+
+import pandas as pd
+import numpy as np
+
+from sklearn.decomposition import PCA
+from sklearn.cluster import KMeans
+from sklearn import metrics
+from sklearn import preprocessing
+from statistics import mode
+
+import s3fs
+import pyarrow.parquet as pq
+
+import mlflow
+import mlflow.sklearn
+
+import ceph
+
+
+def preprocess_rules_data(DF):
+	
+	# Remove ID and time Columns from training set
+	DF = DF.iloc[:,1:-2] 
+	# Convert any remaining textual data into a nan value (there are very few of these in the dataset)
+	DF = DF.apply(lambda L: pd.to_numeric(L,errors='coerce'))
+	# Fill in any remaning nan values with 0. 
+	DF = DF.fillna(0)
+	return DF
+
+def group_clusters(results,num_clusters):
+
+	day = {}
+
+	# initialize our day dictionary with our initial classes
+	for i in range(num_clusters):
+		current_list = []
+		for j in range(len(results)):
+			if i == results[j][1]:
+				current_list.append(results[j][0]) 
+		day[str(i)] = set(current_list)
+
+
+	return day
+
+
+def calculate_stability_score(day_1,day_2,num_clusters,ids_in_both_days):
+
+	stability_score = 0 
+
+	for i in range(num_clusters):
+		smallest_difference = np.inf
+		for j in range(num_clusters):
+			#find best matching cluster:
+			current_difference  = len(day_1[str(i)].difference(day_2[str(j)]))
+			if current_difference < smallest_difference:
+				smallest_difference = current_difference
+		#print(i, " : ", smallest_difference)    
+		stability_score += smallest_difference
+
+	stability_score = ((1-(stability_score/len(ids_in_both_days))) * 100.0 )  
+	#print(str(stability_score) , "%,")
+
+	return stability_score 
+
+
+def run_clustering():
```

Comment:
the clustering should stay in the clustering module. the `metric_tracking` module should only read the clusters via the storage api and calculate the metric

Reply:
I'm copying all of Michael's clustering code over to __init__.py in the clustering directory. What should go in the `metric_tracking` directory?